web/ui/react-app/src/utils/index.ts

```diff
@@ -210,3 +211,11 @@ export const callAll = (...fns: Array<(...args: any) => void>) => (...args: any)
   // eslint-disable-next-line prefer-spread
   fns.filter(Boolean).forEach(fn => fn.apply(null, args));
 };
+
+export const replaceWith = (str: string, needle: string, replace: string) => {
+  return str.replace(new RegExp(needle, 'g'), `${replace}`);
+};
+
+export const isAlertinRule = (rule: Rule): rule is AlertingRule => {
```

Comment:
I don't think it's worth putting this into its own method, but we could e.g. have a string enum type for the rule types, if you prefer not to use direct string comparisons in conditions.